### PR TITLE
Do some code cleanup of COM2 interop classes

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Basic cleanup. Also a lot of removing redundant comments.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7989)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2ComponentEditor.cs [hunk 1]
```diff
@@ -31,12 +31,7 @@ public static unsafe bool NeedsComponentEditor(object obj)
                 try
                 {
                     HRESULT hr = ispp.GetPages(&uuids);
-                    if (!hr.Succeeded)
-                    {
-                        return false;
-                    }
-
-                    return uuids.cElems > 0;
+                    return !hr.Succeeded ? false : uuids.cElems > 0;
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2ComponentEditor.cs:34]:
```suggestion
                    return hr.Succeeded && uuids.cElems > 0;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2ExtendedTypeConverter.cs [hunk 2]
```diff
@@ -59,182 +52,65 @@ public Com2ExtendedTypeConverter([DynamicallyAccessedMembers(DynamicallyAccessed
...
-        /// </summary>
         public override bool GetCreateInstanceSupported(ITypeDescriptorContext? context)
-        {
-            if (_innerConverter is not null)
-            {
-                return _innerConverter.GetCreateInstanceSupported(context);
-            }
+            => _innerConverter is not null
+                ? _innerConverter.GetCreateInstanceSupported(context)
+            : base.GetCreateInstanceSupported(context);
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2ExtendedTypeConverter.cs:83]: nit: indenting looks a little off here

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IPerPropertyBrowsingHandler.cs [hunk 3]
```diff
@@ -38,15 +32,10 @@ public override void SetupPropertyHandlers(Com2PropertyDescriptor[]? propDesc)
...
+            // Check for a property page
             Guid guid = Guid.Empty;
             HRESULT hr = target.MapPropertyToPage(dispid, &guid);
-            if (hr == HRESULT.S_OK)
-            {
-                return guid;
-            }
-
-            return Guid.Empty;
+            return hr == HRESULT.S_OK ? guid : Guid.Empty;
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IPerPropertyBrowsingHandler.cs:38]:
```suggestion
            return hr.Succeeded ? guid : Guid.Empty;
```

**Review by lonitra — APPROVED** [review]
👍

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IDispatchConverter.cs [hunk 4]
```diff
@@ -29,77 +29,41 @@ public Com2IDispatchConverter(Com2PropertyDescriptor propDesc, bool allowExpand)
...
+            string text = ComNativeDescriptor.GetName(value);
 
-                return text;
+            if (string.IsNullOrEmpty(text))
+            {
+                text = ComNativeDescriptor.GetClassName(value);
             }
 
-            return base.ConvertTo(context, culture, value, destinationType);
+            return text is null ? "(Object)" : (object)text;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IDispatchConverter.cs:56]: is this conversion to `object` needed?

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IDispatchConverter.cs (hunk 4 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IDispatchConverter.cs:56]: It isn't- I used auto-fixes for these and it doesn't remove things like this. I'll catch it in my next change.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2Properties.cs [hunk 5]
```diff
@@ -277,44 +206,41 @@ public void AddExtendedBrowsingHandlers(Hashtable handlers)
                 return;
             }
 
-            // process all our registered types
+            // Process all our registered types.
             Type t;
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2Properties.cs:210]: I'll look closer for them with my coming work.

**Pull request merged**